prompt2model/prompt_parser/instr_parser.py

```diff
@@ -100,29 +95,31 @@ def parse_from_prompt(self, prompt: str) -> None:
 
         chat_api = ChatGPTAgent(self.api_key)
         while True:
-            try:
-                self.api_call_counter += 1
-                response = chat_api.generate_one_openai_chat_completion(
-                    parsing_prompt_for_chatgpt,
-                    temperature=0,
-                    presence_penalty=0,
-                    frequency_penalty=0,
-                )
-                extraction = self.extract_response(response)
-                if extraction is not None:
-                    self._instruction, self._examples = extraction
-                    return None
-                else:
-                    if (
-                        self.max_api_calls
-                        and self.api_call_counter == self.max_api_calls
-                    ):
-                        logger.warning(
-                            "Maximum number of API calls reached for PromptParser."
-                        )
-                        return None
-            except OPENAI_ERRORS as e:
-                self.api_call_counter = handle_openai_error(e, self.api_call_counter)
+            self.api_call_counter += 1
+            response = chat_api.generate_one_openai_chat_completion(
+                parsing_prompt_for_chatgpt,
+                temperature=0,
+                presence_penalty=0,
+                frequency_penalty=0,
+            )
+
+            if isinstance(response, Exception):
```

Comment:
I had a fair amount of trouble understanding the control flow of the rest of this function. I wonder if it could be simplified and/or commented about what all of these branches mean.

Reply:
Very valid point!
- I have reduced a lot of branching that was unnecessary from the remainder of the function
- I have updated the comments and docstrings to accurately reflect the flow.

Let me know if it is still unclear